### PR TITLE
[semver:minor] Make validating dependencies optional

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
Validating dependencies with `dependency:go-offline` in multi-package projects that have interdependencies will fail when updating the version numbers. This is because the project has not yet been built and the dependencies are not available from public sources.

This changes updates the `with_cache` command to optionally execute the `dependency:go-offline` goal based on a parameter. The default is for the dependencies to be checked and allows for the them to be omitted using a new _verify_dependencies_ parameter.

### CODE REVIEW

**Comment by skjolber** [comment]
@KyleTryon @gmemstr what is the holdup? Customers (like us) really expect basic stuff like multi-module builds to be working.

**Comment by KyleTryon** [comment]
> @KyleTryon @gmemstr what is the holdup? Customers (like us) really expect basic stuff like multi-module builds to be working.

Apologies, it doesn't look as though we were pinned on the PR and may not have seen it as the two of us are not maven experts. It appears there is a small conflict to be corrected but otherwise sounds good. The default setting appears to be in line with the current behavior and so with that conflict correction, this should be a valid minor release.

**Comment by KyleTryon** [comment]
This Orb is still currently using the older development pipeline (Pre orb-tools 10.0), I'd like to update the dev pipeline separately, we can include this feature as well. Leaving this open until the new PR is ready.

**Comment by skjolber** [comment]
Hope updating the development pipeline is quick; otherwise it would be fair to fast-track this as a critical issue (Maven being the main build tool for one of the central ecosystems and all).

**Comment by MikeGoldsmith** [comment]
@KyleTryon are you wanting me to resolve conflicts so we can merge this PR or is there something else?

**Comment by KyleTryon** [comment]
> @KyleTryon are you wanting me to resolve conflicts so we can merge this PR or is there something else?

If you wouldn't mind confirming 👍, I'll re-trigger the tests and if need-be ill manually deploy this so we can get it out faster / before we update the pipeline.

**Comment by skjolber** [comment]
@KyleTryon Please take ownership of this change if the original submitter is not responding, as soon as possible.

**Comment by MikeGoldsmith** [comment]
I can make the changes, I didn't realise you were waiting on me. 

I'll update my branch

**Comment by MikeGoldsmith** [comment]
@KyleTryon I've resolved the outstanding conflicts, I think this is ready for final review now 👍

File: src/commands/with_cache.yml [hunk 1]
```diff
@@ -26,6 +26,10 @@ parameters:
     description: Useful when the source of your maven project is nott in the root directory of your git repo. Supply the name of the directory or relative path of the directory containing your source code.
     type: string
     default: ''
+  verify_dependencies:
+    description: Verify dependencies are valid and available from pubic sources
```
Comment (KyleTryon) [inline src/commands/with_cache.yml:30]: public* lol

File: src/commands/with_cache.yml [hunk 2]
```diff
@@ -40,7 +44,11 @@ steps:
         if [ -n "<< parameters.settings_file >>" ]; then
           set -- "$@" --settings "<< parameters.settings_file >>"
         fi
-        << parameters.maven_command >> org.apache.maven.plugins:maven-dependency-plugin:3.1.2:go-offline "$@"
+  - run:
+      name: Verify dependencies
+      working_directory: << parameters.app_src_directory >>
+      when: << parameters.verify_dependencies >>
```
Comment (KyleTryon) [inline src/commands/with_cache.yml:50]: Improper use of when: https://circleci.com/docs/2.0/configuration-reference/#the-when-attribute

You may instead be looking for this: https://circleci.com/docs/2.0/configuration-reference/#the-when-step-requires-version-21

**Review by KyleTryon — CHANGES_REQUESTED** [review]

File: src/commands/with_cache.yml (hunk 2 above)
Comment (MikeGoldsmith) [reply src/commands/with_cache.yml]: @KyleTryon I've updated to use the latter type, let me know if that looks right.

**Comment by skjolber** [comment]
@KyleTryon all good now?

**Comment by KyleTryon** [comment]
Thank you all, the change has been merged and will now deploy

**Pull request merged**